community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/TemporalIndexCache.java

```diff
@@ -99,31 +104,24 @@ T uncheckedSelect( ValueGroup valueGroup )
     }
 
     /**
-     * Select the part corresponding to the given ValueGroup. Creates the part if needed,
+     * Select the getOrCreatePart corresponding to the given ValueGroup. Creates the getOrCreatePart if needed,
      * in which case an exception of type E might be thrown.
      *
      * @param valueGroup target value group
-     * @return selected part
+     * @return selected getOrCreatePart
      */
     T select( ValueGroup valueGroup ) throws IOException
     {
-        try
-        {
-            return uncheckedSelect( valueGroup );
-        }
-        catch ( UncheckedIOException e )
-        {
-            throw e.getCause();
```

Comment:
Why do you remove this catch clause and why is it left intact in `SpatialIndexCache`?

Reply:
I think this was a mistake. I started changing how exceptions were wrapped inside the lambdas, but had to back out and missed this one.